src/EnergyPlus/DaylightingManager.cc

```diff
@@ -3230,41 +3171,42 @@ void FigureDayltgCoeffsAtPointsForSunPosition(
             // Make all transmitted light diffuse for a TDD with a bare diffuser
             assert(equal_dimensions(state.dataDaylightingManager->AVWLSK, state.dataDaylightingManager->WLUMSK));
             assert(equal_dimensions(state.dataDaylightingManager->AVWLSK, state.dataDaylightingManager->EDIRSK));
-            int l3(state.dataDaylightingManager->AVWLSK.index(iHour, 1, 1));
-            for (ISky = 1; ISky <= 4; ++ISky, ++l3) { // [ l3 ] == ( ISky, 1, iHour )
-                state.dataDaylightingManager->AVWLSK[l3] += state.dataDaylightingManager->WLUMSK[l3];
-                if (ISky == 1) {
-                    state.dataDaylightingManager->AVWLSU(iHour, 1) += state.dataDaylightingManager->WLUMSU(iHour, 1);
-                    state.dataDaylightingManager->AVWLSUdisk(iHour, 1) += state.dataDaylightingManager->WLUMSUdisk(iHour, 1);
-                }
-                if (PHRAY > 0.0) {
-                    state.dataDaylightingManager->EDIRSK[l3] += state.dataDaylightingManager->WLUMSK[l3] * DOMEGA_Ray_3;
-                    if (ISky == 1) state.dataDaylightingManager->EDIRSU(iHour, 1) += state.dataDaylightingManager->WLUMSU(iHour, 1) * DOMEGA_Ray_3;
-                }
+            auto &avwlsk = state.dataDaylightingManager->AVWLSK(iHour, 1);
+            auto &edirsk = state.dataDaylightingManager->EDIRSK(iHour, 1);
+            auto &wlumsk = state.dataDaylightingManager->WLUMSK(iHour, 1);
+            for (int iSky = (int)SkyType::Clear; iSky < (int)SkyType::Num; ++iSky) { 
+                avwlsk.sky[iSky] += wlumsk.sky[iSky];
+                if (PHRAY > 0.0) 
+                    edirsk.sky[iSky] += wlumsk.sky[iSky] * DOMEGA_Ray_3;
             }
 
+            state.dataDaylightingManager->AVWLSU(iHour, 1) += state.dataDaylightingManager->WLUMSU(iHour, 1);
```

Comment:
Something else I did was pull out all of the `if (ISky == 1) {}` to unconditional code outside the loop.